lib/network/kmip/ClusterClient.ts

```diff
@@ -270,7 +270,23 @@ export default class ClusterClient implements KMSInterface {
     }
 
     healthcheck(...args: Parameters<Required<KMSInterface>['healthcheck']>) {
-        // for now check health of every member
-        this.clusterHealthcheck.apply(this, args);
+        // healthcheck follow round robin with retries instead of broadcast
+        const originalCallback = args.pop() as ArrayLast<typeof args>;
+        const poppedArgs = args as unknown as ArrayPopped<typeof args>;
+        const logger = args[args.length - 1] as ArrayLast<typeof poppedArgs>;
+
+        if (this.clients.length === 0) {
+            logger.warn('kmip cluster has no healthy hosts');
+            return originalCallback(errorInstances.InternalError.customizeDescription(
+                kmipMsg('Healthcheck', '', `no healthy host in the cluster`)));
+        }
+
+        const client = this.next();
+
+        client.healthcheck(
```

Comment:
Are we sure the caller will retry enough times to ensure our definition of "healthy" is satisfied? Otherwise, it may be necessary to retry in this function in a round-robin fashion as you state in the comment.

Reply:
`this.callback` takes care of retrying.

Healthcheck will now act like any other action, it retries until the configured `retries` is reached, which by default will retry on every hosts of the cluster until a success or they are all marked unhealthy.

But if the client configure a smaller `retries` number then it might not go over all hosts. 

But we'll rework the healthcheck for 9.5.1 so we might provide different ways to do later.